### ISSUE TITLE
Unicode rendering odd behavior

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mintty/mintty#285]
```
What steps will reproduce the problem?
1. Echo 'ಠ_ಠ' to the console using 'Anonymous Pro (TT)' font (freely 
available).

What is the expected output? What do you see instead?
The underscore character appears shorter than usual, about half width.

What versions of mintty, Cygwin/MSYS, and Windows are you using?
Windows 7 Enterprise x64
Cygwin latest (don't know how to determine version; don't think it's relevant 
anyhow, since this is a rendering issue)
Mintty 1.0.1

Please provide any additional information below.
Selecting one of the ಠ characters appears to fix the rendering of the 
underscore character to full width, but only temporarily. It reverts to the 
shortened underscore when deselecting.

The attached screenshot shows the defect on the `$ echo` line. The line below 
with the inverted character shows me selecting that character where the 
rendering of the underscore next to it is corrected.
```

Original issue reported on code.google.com by `james.jdunne@gmail.com` on 4 Aug 2011 at 9:34

Attachments:
* [unicode-font-switch-defect.png](https://storage.googleapis.com/google-code-attachments/mintty/issue-285/comment-0/unicode-font-switch-defect.png)

**Comment by mperry2** [comment]
No response from the original reporter in almost five years. Should this be closed?

**Comment by mintty** [comment]
Thanks for caring, but no, this is an issue and there are more examples of glyphs flipping in appearance once you move the cursor over them (in an editor):
	"xفษษx刈"ษ
I've not have time to analyse it, and I'm not sure whether mintty will have the capability to fix this or maybe it's a Windows issue.

**Comment by mintty** [comment]
Released fix in 2.3.7.

**Comment by mintty** [comment]
Also, digits used to be adapted to the surrounding script style (?), like
x841
x841 ꡁ

That's also fixed now.

**Comment by mintty** [comment]
Released further fixes and tweaks in 2.4.0.
